resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -279,7 +299,13 @@ private[spark] class KubernetesClusterSchedulerBackend(
     }
   }
 
-  private def allocateNewExecutorPod(): (String, Pod) = {
+  /**
+   * Allocates a new executor pod
+   * @param nodeToTaskCount A map of K8s cluster nodes to the number of tasks that could benefit
+   *                        from data locality if an executor launches on the cluster node.
+   * @return A tuple of the new executor name and the Pod data structure.
+   */
+  private def allocateNewExecutorPod(nodeToTaskCount: Map[String, Int]): (String, Pod) = {
```

Comment:
this method has grown to >150 lines -- maybe a followup PR can shrink it by refactoring in some way?